### PR TITLE
Add choose one polls

### PR DESCRIPTION
**Pull request by pietgeursen**
selection_636

Try it out now ! 
1. pull this branch down
2. npm install
3. push the buttons!

Todo:

- [x] i18n
- [x] display mini cards in the main feed

Todo (later)
- [ ] Make ui better so you can't double submit a poll / position
- [ ] Display create poll form errors
- [ ] Stop user from voting on a closed poll
- [ ] Display positions that are invalid(late, incorrect choice etc) as errors in the poll results
- [ ] filter by polls you are participating in
- [ ] so many things!...

### CODE REVIEW

File: main-window.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ module.exports = function (config) {
   })
 
   var views = api.app.views(api.page.html.render, [
-    '/public', '/private', id, '/mentions'
+    '/polls', '/public', '/private', id, '/mentions'
```
Comment (mixmix) [inline main-window.js]: remove this before merging!

File: modules/feed/html/rollup.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ var bumpMessages = {
   'about': 'added changes',
   'mention': 'mentioned you',
   'channel-mention': 'mentioned this channel',
-  'attending': 'can attend'
+  'attending': 'can attend',
+  'participated': 'participated in this poll'
```
Comment (mixmix) [inline modules/feed/html/rollup.js]: make the key more poll specific

**Comment by Serkan-devel** [comment]
Are there also pools with multiple choices?

**Comment by mixmix** [comment]
Serken we built a module which will handle about 5 different type of polls.
We are you offish in bringing one type completely through before building
others.
So yes!

File: main-window.js (hunk 1 above)
Comment (mixmix) [reply main-window.js]: actually having this entry be `process.env.START_PAGE` would be handy
I have something like this in Ticktack

File: modules/poll/sheet/edit.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        h('PollEditor', [
+          h('input.title', { 'ev-input': ev => poll.title.set(ev.target.value), placeholder: i18n('Choose a title') }, poll.title),
+          h('textarea.description', { 'ev-input': ev => poll.body.set(ev.target.value), placeholder: i18n('Describe the poll if you want') }, poll.body),
+
+          h('div.field -choices', [
+            h('div.inputs', [
+              map(poll.choices, (choice, index) => {
+                return h('input', { 'ev-input': ev => choice.set(ev.target.value), placeholder: i18n('Choice') }, choice)
+              }),
+              h('button', { 'ev-click': () => poll.choices.push(Value()) }, i18n('+ Add more choices'))
```
Comment (ahdinosaur) [inline modules/poll/sheet/edit.js]: user feedback: it felt weird to start with 3 empty choices and not be able to remove any of them.

File: modules/poll/sheet/edit.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            ])
+          ]),
+          h('input.date', {
+            placeholder: i18n('Choose date and time'),
```
Comment (ahdinosaur) [inline modules/poll/sheet/edit.js]: user feedback: what is this date and time for? is it when the poll closes?

File: plugs/message/html/render/poll.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+            )
+          ]),
+          h('div.field -reason', [
+            h('label', 'Reason'),
+            h('textarea', { 'ev-input': ev => newPosition.reason.set(ev.target.value) }, newPosition.reason)
+          ]),
+          h('div.actions', [
+            h('button.publish.-primary', { 'ev-click': publish }, 'Publish position')
+          ]),
+          h('div.changePosition', { 'ev-click': ev => forceShow.set(true) },
```
Comment (ahdinosaur) [inline plugs/message/html/render/poll.js]: user feedback: felt weird to be missing a cancel button, what if i don't really want to change my position?

**Comment by ahdinosaur** [comment]
oh also i was able to vote in a closed poll (`%Z4h9cmCzSSxPsNnKlQuk+wvu44YDdIbjuH5rOfGPHgY=.sha256`), it seems to show everything as if i'm a normal participant of the poll, not sure if that's expected or what.

File: modules/poll/sheet/edit.js (hunk 4 above)
Comment (mixmix) [reply modules/poll/sheet/edit.js]: yes

File: modules/poll/sheet/edit.js (hunk 3 above)
Comment (mixmix) [reply modules/poll/sheet/edit.js]: good point. I know that it tidies under the hood if there are empty fields, but that's not indicated at all.

what would you prefer or suggest?

File: plugs/message/html/render/poll.js (hunk 5 above)
Comment (mixmix) [reply plugs/message/html/render/poll.js]: nice catch

**Comment by mixmix** [comment]
hey @mmckegg would love your input on this PR (more than just having it WIP'd).

There's little bits of feedback I want to fold in but I want to wait till you have capacity and we can check in about what you want to get this merged. I'm also happy for you to take over and do whatever you want with this and merge it. The point is to get some poll's in as many clients as possible

**Comment by mmckegg** [comment]
Initial thoughts:

1. Wrap cards inside of standard message layout (like gatherings)
2. make filter buttons navigate to urls rather than using state
3. Figure out how to integrate positions and posts into thread
  - hide entire thread before voted on

File: modules/page/html/render/polls.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+exports.create = function (api) {
+  const i18n = api.intl.sync.i18n
+
+  return nest('page.html.render', function channel (path) {
+    if (path !== '/polls') return
+
+    var scuttlePoll = ScuttlePoll(api.sbot.obs.connection)
+    var id = api.keys.sync.id()
+
+    var mode = Value(OPEN)
```
Comment (mixmix) [inline modules/page/html/render/polls.js]: this is state which leads to page being changed but route note being changed.

matt wants this to use routes to do the changes instead .. i.e. those filters should go to : 
- /polls  (same as /polls/open ?)
- /polls/open
- /polls/closed
- /polls/all

**Comment by mixmix** [comment]
and **4. i18n**

**Comment by pietgeursen** [comment]
Okay @mmckegg 

- [x]    Wrap cards inside of standard message layout (like gatherings)
- [x]   make filter buttons navigate to urls rather than using state
- [x]   Figure out how to integrate positions and posts into thread

**Comment by mmckegg** [comment]
Hey @pietgeursen,

Sorry, yes keen to merge. Just need to find the time to review etc. I think me and @mixmix talked about peering on this sometime this week.

**Comment by mmckegg** [comment]
@pietgeursen 

Just having a look at this now (and merged the latest master in). I see there are still things missing (styles, closed polls etc), so not ready to merge.

Are you wanting me to review the new code changes?

**Comment by mmckegg** [comment]
Hmm, maybe my merge broke some stuff? A bunch of styles seem to be missing (including all dark ones).

(this looks okay)
screen shot 2018-08-22 at 8 16 44 pm

(but the rest...)
screen shot 2018-08-22 at 8 16 37 pm
screen shot 2018-08-22 at 8 32 18 pm
screen shot 2018-08-22 at 8 32 48 pm

**Comment by mixmix** [comment]
Yeah i want to get this done. It's hurting having it open

**Pull request closed without merging**